integration/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/common/ServerConfigurationManager.java

```diff
@@ -310,6 +310,24 @@ public void applyMIConfiguration(File newConfig) throws AutomationUtilException,
         restartMicroIntegrator();
     }
 
+    /**
+     * apply configuration file and optionally restart micro integrator server to take effect the configuration
+     *
+     * @param newConfig       configuration file
+     * @param restartRequired optional Restart flag
+     * @throws AutomationUtilException - throws if apply configuration fails
+     * @throws IOException             - throws if apply configuration fails
+     */
+    public void applyMIConfiguration(File newConfig, boolean restartRequired)
```

Comment:
Shall we remove the other method without overloading. Otherwise it would be used un intentionally

Reply:
or keep it protected